lib/hashes.md

```diff
@@ -5,40 +5,40 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
 # What is a Hash?
 
-Replace me with your response...
+A hash is an unordered list of key value pairs.  Other languages call it a dictionary sometimes and that's not such a horrible way to think about them.  It's similar to an array, but the key value pairs are unique and use that for indexing, as opposed ot an array which is indexed using integers.  
 
 # What are some examples of information that would be Hashes as opposed to some other data type?
 
-Replace me with your response...
+Well, as mentioned, other languages call hashes dictionaries.  Matching a word to a definition isn't so far off from matching a key to a value.  You could use a hash to keep track of paired bits of data, e.g., a phone book pairing names to numbers, a gradebook pairing students to their grades, or keeping inventory pairing products with their quantities or prices.
 
 # How are Hashes and Arrays similar? How are they different?
 
-Replace me with your response...
+Well, superficially, they're indicated differently in ruby.  Whereas brackets [] were the order of the day with arrays, hashes are indicated by braces {}.  So, if an array would look like array_1 = [1,2,3,4,5] whereas a hash would look like hash_1 = { "Suzie" => 10, "Calvin" => 2, "Hobbes" => 7}  Beyond that, as mentioned, they're indexed differently.  An array is indexed using integers, so if you want to call up a particular element in an array, you'd indicate its position,e.g., in array_1 = [1,2,3,4,5] each item has a position number and to call up that item you look up that position.  For a hash, it's indexed via the unique keys, thus in hash_1 = ["Suzie" => 10, "Calvin" => 2, "Hobbes" => 7} you'd indicate the key to call up its corresponding value. So, they're similar to one another in that they're both collections of data, but that data is structured and retrieved in different manners and better suited to different tasks.
 
 # How do you retrieve a particular value from a Hash?
 
-Replace me with your response...
+To retrieve a particular value from a hash, you'd call up it's key.  So in hash_1, if you wanted to call up the value associated with "Suzie" you'd say hash_1["Suzie"] and it should return 10.
 
 # How do you add information to a Hash?
 
-Replace me with your response...
+You should be able to add a new key value pair by entering them in the format hash[key] = value, where "hash" is whatever it is you've named your hash.
 
 # How would you perform an operation on every element inside a Hash?
 
-Replace me with your response...
+You'd use the each method.  So, if you had a hash wherein you wanted to increase all your values (assuming they're integers) by a value of 1, you could type something like hash.each do |key, value| and specify in the block hash[key] = value  + 1.  It's worth noting that this change is superficial.  It doesn't actually change the values in the hash, it just performs a given action or set of actions on them.
```

Comment:
> It's worth noting that this change is superficial.  It doesn't actually change the values in the hash, it just performs a given action or set of actions on them.

This statement warrants clarification. Let's say that you run the following code:

```ruby
hash = {a: 1, b: 5, c: 6}

hash.each do |key, value|
  hash[key] = value + 1
end
```

After this code is run, if we check what `hash` is, we'll find that it _has_ changed from `{a: 1, b: 5, c: 6}` to `{a: 2, b: 6, c: 7}`.

But if we run the following code:

```ruby
hash = {a: 1, b: 5, c: 6}

hash.each do |key, value|
  puts value + 1
end
```

Then `hash` remains unchanged.

Can you think about this, and ask questions if needed? Then please elaborate/clarify your original statement that "It's worth noting that this change is superficial.  It doesn't actually change the values in the hash, it just performs a given action or set of actions on them."